### PR TITLE
refactor: post comments query to allow sorting

### PR DESCRIPTION
**Pull request by sshanzel**

### CODE REVIEW

**Comment by sshanzel** [comment]
@dailydotdev/web-team anyone know why I'm suddenly getting an error that it "is not a function"?

Screenshot 2024-09-13 at 6 59 35 PM

File: src/schema/comments.ts [hunk 1]
```diff
@@ -98,7 +99,14 @@ export interface GQLUserComment {
   votedAt: Date | null;
 }
 
+export enum SortCommentsBy {
+  NewestFirst = 'newest',
+  OldestFirst = 'oldest',
+}
+
 export const typeDefs = /* GraphQL */ `
+  ${toGQLEnum(SortCommentsBy, 'SortCommentsBy')}
```
Comment (capJavert) [inline src/schema/comments.ts:108]: @sshanzel you probably have circual dependency

Comment (sshanzel) [reply src/schema/comments.ts:108]: Yeah, was thinking that too, but wanted to confirm. Let me check this one quickly 👀

Comment (capJavert) [reply src/schema/comments.ts:108]: fix it by importing from `import { toGQLEnum } from '../common/utils'`

Comment (capJavert) [reply src/schema/comments.ts:108]: Basically try to avoid barrel files

Comment (sshanzel) [reply src/schema/comments.ts:108]: Yikes! Never knew doing that would be such an issue. Worked now. Thanks!

**Review by capJavert — APPROVED** [review]

**Review by nensidosari — APPROVED** [review]

**Pull request merged**